compiler/luci/lang/include/luci/IR/ProfilingData.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <loco.h>
+
+#include <set>
+
+namespace luci
+{
+
+/**
+ * @brief Set annotation for Profiling data
+ */
+class ProfilingData final : public loco::NodeAnnotation
+{
+public:
+  ProfilingData()
+  {
+    // DO NOTHING
+  }
+
+  ProfilingData(int32_t id)
+  {
+    _node_id = id;
+    _origins.insert(id);
+  }
+
+  ProfilingData(uint32_t id)
+  {
+    _node_id = static_cast<int32_t>(id);
+    _origins.insert(static_cast<int32_t>(id));
+  }
```

Comment:
(This is not related to backend perspective but...)

These ctors looks strange to me. I guess there is no relation between node id and origins

Reply:
Sorry for my poor design skill :(
At first, it would be better to include both of them but it would be better to divide `CircleNodeID` and `CircleNodeOrigin`